producer/src/main/kotlin/streams/procedures/StreamsProcedures.kt

```diff
@@ -15,15 +40,15 @@ class StreamsProcedures {
     @Procedure(mode = Mode.SCHEMA, name = "streams.publish")
     @Description("streams.publish(topic, config) - Allows custom streaming from Neo4j to the configured stream environment")
     fun publish(@Name("topic") topic: String?, @Name("payload") payload: Any?,
-                @Name(value = "config", defaultValue = "{}") config: Map<String, Any>?) {
+                @Name(value = "config", defaultValue = "{}") config: Map<String, Any>?): Stream<PublishResult> {
         checkEnabled()
         if (topic.isNullOrEmpty()) {
             log?.info("Topic empty, no message sent")
-            return
+            return Stream.empty();
         }
         if (payload == null) {
             log?.info("Payload empty, no message sent")
-            return
+            throw RuntimeException("Payload may not be null")
```

Comment:
don't do that. imagine you have many messages then one null will result in a half-send batch